src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedIntoPrisonEventService.kt

```diff
@@ -2,30 +2,61 @@ package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.dto.CreateInductionScheduleDto
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.service.InductionService
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation.Reason.ADMISSION
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation.Reason.RETURN_FROM_COURT
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation.Reason.TEMPORARY_ABSENCE_RETURN
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation.Reason.TRANSFERRED
+import java.time.LocalDate
+import java.time.ZoneId
 
 private val log = KotlinLogging.logger {}
 
 @Service
-class PrisonerReceivedIntoPrisonEventService {
-  fun process(inboundEvent: InboundEvent, additionalInformation: PrisonerReceivedAdditionalInformation) =
+class PrisonerReceivedIntoPrisonEventService(private val inductionService: InductionService) {
+  fun process(inboundEvent: InboundEvent, additionalInformation: PrisonerReceivedAdditionalInformation) {
     when (additionalInformation.reason) {
       ADMISSION -> {
-        log.info { "Processing Prisoner Received Into Prison Event with reason ${additionalInformation.reason}" }
-        // TODO - process prisoner admission event in respect of what PLP needs to do on this event
+        log.info { "Processing Prisoner Received Into Prison Event with reason ${additionalInformation.reason} and prisoner number ${inboundEvent.prisonNumber()}" }
+        try {
+          inductionService.createInductionSchedule(
+            CreateInductionScheduleDto(
+              deadlineDate = calculateDeadlineDate(inboundEvent),
+              prisonNumber = inboundEvent.prisonNumber(),
+              scheduleCalculationRule = InductionScheduleCalculationRule.NEW_PRISON_ADMISSION,
+            ),
+          )
+        } catch (inductionAlreadyExistsException: InductionAlreadyExistsException) {
+          log.info { "An induction already exists for prisoner with id ${inboundEvent.prisonNumber()}" }
+          // update existing deadline date
+          // TODO
+        } catch (inductionScheduleAlreadyExistsException: InductionScheduleAlreadyExistsException) {
+          log.info { "An induction schedule exists for prisoner with id ${inboundEvent.prisonNumber()}" }
+        }
       }
 
       TRANSFERRED -> {
         log.info { "Processing Prisoner Received Into Prison Event with reason ${additionalInformation.reason}" }
+        // Should already have an induction but Create induction or update existing deadline date if one exists
```

Comment:
I think the logic here might be identical to whatever we write for the Admission event above 🤔 
I _think_ in terms of whether they have an Induction and/or InductionSchedule, and what we need to do .... I think it's the same 🤔 
We should check that assumption though